### PR TITLE
Many annotation fixes, pydetector dependency upgraded

### LINKED ISSUE #63: Decorators have the same line number as the functions they decorate
Hi,
I tested https://github.com/EgorBu/uast_playground at a new file
`python3 -m uast_playground repo2id_str -r test_data/exp2`

screenshot from 2017-08-10 11-02-10

1) wrong start and end positions of tokens (token has more than 1 line, overlaps, wrong length, etc)
3) decorators and function definitions have the same start line

### LINKED ISSUE #65: Snippet produces wrong offset
This code:

```python
class Repo2nBOW(Repo2Base):
    @property
    def id2vec(self):
        return self._id2vec
```

Produces and offset of 0 for `_id2vec`.

### PR DESCRIPTION
**Pull request by juanjux**
- Upgrade pydetector dependency to 0.12.0 (which fixes several positioning problems).
- Added regression tests for #63 and #65.
- Many annotations fixes:
  - Fixed import annotations and identifiers.
  - Remove several unneded promoted lists.
  - Moved some annotations out of child lists to the top level since they were the
    same under all the parents.
  - Added Incomplete to ClassDef keywords (`metaclass=bla`, etc).
  - Removed temporal files and directories from the integration tests that were in git.

Fixes #63
Fixes #65

### CODE REVIEW

File: ANNOTATION.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+| Path | Action |
```
Comment (abeaumont) [inline ANNOTATION.md]: oh, this file was removed? :(

File: tests/import.py.uast [hunk 2]
```diff
@@ -30,29 +30,22 @@ Module {
...
-.  .  .  .  .  .  .  EndPosition: {
-.  .  .  .  .  .  .  .  Offset: 9
-.  .  .  .  .  .  .  .  Line: 1
-.  .  .  .  .  .  .  .  Col: 10
-.  .  .  .  .  .  .  }
-.  .  .  .  .  .  .  Properties: {
-.  .  .  .  .  .  .  .  asname: <nil>
-.  .  .  .  .  .  .  }
-.  .  .  .  .  .  }
+.  .  .  .  .  .  asname: <nil>
```
Comment (abeaumont) [inline tests/import.py.uast]: Is this expected?

File: ANNOTATION.md (hunk 1 above)
Comment (juanjux) [inline ANNOTATION.md]: Not removed but the last release left it empty for some reason...

File: tests/import.py.uast (hunk 2 above)
Comment (juanjux) [inline tests/import.py.uast]: Yes, is in the part of the native properties of the node and is `None` in Python (is the property that maps "y" in "import x as y", it's always present even for `import x` without alias).

**Pull request merged**